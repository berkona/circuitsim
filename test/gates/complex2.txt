{"undoStack":[],"graph":{"0":{"nid":"0","type":"input","name":"A","pos":{"x":33.82500457763672,"y":51.743743896484375},"rect":{"x":33.82500457763672,"y":51.743743896484375,"width":24,"height":24},"pins":[{"adj":[["4","1"],["28","1"]]}]},"1":{"nid":"1","type":"vcc","pos":{"x":80.10000610351562,"y":2},"rect":{"x":80.10000610351562,"y":2,"width":40,"height":42},"pins":[{"adj":[["55",0]]}]},"2":{"nid":"2","type":"input","name":"B","pos":{"x":29.82500457763672,"y":99.7437744140625},"rect":{"x":29.82500457763672,"y":99.7437744140625,"width":24,"height":24},"pins":[{"adj":[["5","1"],["30","1"]]}]},"4":{"nid":"4","type":"pmos","pos":{"x":83.5999984741211,"y":60},"rect":{"x":83.5999984741211,"y":60,"width":40,"height":43},"pins":[{"adj":[["56","0"]]},{"adj":[["0","0"]]},{"adj":[["5","0"]]}]},"5":{"nid":"5","type":"pmos","pos":{"x":79.5999984741211,"y":114},"rect":{"x":79.5999984741211,"y":114,"width":40,"height":43},"pins":[{"adj":[["4","2"]]},{"adj":[["2","0"]]},{"adj":[["21",0]]}]},"6":{"nid":"6","type":"input","name":"E","pos":{"x":183.82500457763672,"y":62.7437744140625},"rect":{"x":183.82500457763672,"y":62.7437744140625,"width":24,"height":24},"pins":[{"adj":[["10","1"],["37","1"]]}]},"7":{"nid":"7","type":"input","name":"F","pos":{"x":179.0374984741211,"y":117.7437744140625},"rect":{"x":179.0374984741211,"y":117.7437744140625,"width":24,"height":24},"pins":[{"adj":[["11","1"],["39","1"]]}]},"10":{"nid":"10","type":"pmos","pos":{"x":221.5999984741211,"y":68},"rect":{"x":221.5999984741211,"y":68,"width":40,"height":43},"pins":[{"adj":[["16","0"]]},{"adj":[["6","0"]]},{"adj":[["11","0"]]}]},"11":{"nid":"11","type":"pmos","pos":{"x":211.5999984741211,"y":120},"rect":{"x":211.5999984741211,"y":120,"width":40,"height":43},"pins":[{"adj":[["10","2"]]},{"adj":[["7","0"]]},{"adj":[["21",0]]}]},"12":{"nid":"12","type":"input","name":"C","pos":{"x":340.59375,"y":81.74375915527344},"rect":{"x":340.59375,"y":81.74375915527344,"width":24,"height":24},"pins":[{"adj":[["14","1"],["44","1"]]}]},"13":{"nid":"13","type":"input","name":"D","pos":{"x":436.59375,"y":79.74375915527344},"rect":{"x":436.59375,"y":79.74375915527344,"width":24,"height":24},"pins":[{"adj":[["15","1"],["49","1"]]}]},"14":{"nid":"14","type":"pmos","pos":{"x":371.5999984741211,"y":59.00001525878906},"rect":{"x":371.5999984741211,"y":59.00001525878906,"width":40,"height":43},"pins":[{"adj":[["25",0]]},{"adj":[["12","0"]]},{"adj":[["22",0]]}]},"15":{"nid":"15","type":"pmos","pos":{"x":465.5999984741211,"y":67.00001525878906},"rect":{"x":465.5999984741211,"y":67.00001525878906,"width":40,"height":43},"pins":[{"adj":[["19","0"]]},{"adj":[["13","0"]]},{"adj":[["22",0]]}]},"16":{"nid":"16","type":"wire","pos":{"x":261,"y":40},"rect":{"x":253,"y":32,"width":16,"height":16},"pins":[{"adj":[["10","0"],["56","0"]]}]},"19":{"nid":"19","type":"wire","pos":{"x":507,"y":58},"rect":{"x":499,"y":50,"width":16,"height":16},"pins":[{"adj":[["15","0"],["25",0]]}]},"20":{"nid":"20","type":"wire","pos":{"x":457,"y":192},"rect":{"x":449,"y":184,"width":16,"height":16},"pins":[{"adj":[["21",0],["22",0]]}]},"21":{"nid":"21","type":"wire","pos":{"x":187,"y":162},"rect":{"x":179,"y":154,"width":16,"height":16},"pins":[{"adj":[["5","2"],["11","2"],["20",0]]}]},"22":{"nid":"22","type":"wire","pos":{"x":455,"y":109.00001525878906},"rect":{"x":447,"y":101.00001525878906,"width":16,"height":16},"pins":[{"adj":[["14","2"],["15","2"],["20","0"]]}]},"23":{"nid":"23","type":"output","name":"Y","pos":{"x":665.8250045776367,"y":167.74374389648438},"rect":{"x":665.8250045776367,"y":167.74374389648438,"width":24,"height":24},"pins":[{"adj":[["24","0"]]}]},"24":{"nid":"24","type":"wire","pos":{"x":645,"y":20},"rect":{"x":637,"y":12,"width":16,"height":16},"pins":[{"adj":[["23","0"],["54",0]]}]},"25":{"nid":"25","type":"wire","pos":{"x":465,"y":58},"rect":{"x":457,"y":50,"width":16,"height":16},"pins":[{"adj":[["14","0"],["19","0"],["54",0]]}]},"26":{"nid":"26","type":"gnd","pos":{"x":35.350006103515625,"y":470.38751220703125},"rect":{"x":35.350006103515625,"y":470.38751220703125,"width":40,"height":36},"pins":[{"adj":[["33",0]]}]},"28":{"nid":"28","type":"nmos","pos":{"x":58.850006103515625,"y":400},"rect":{"x":58.850006103515625,"y":400,"width":40,"height":43},"pins":[{"adj":[["41",0]]},{"adj":[["0","0"]]},{"adj":[["34",0]]}]},"30":{"nid":"30","type":"nmos","pos":{"x":160.85000610351562,"y":393},"rect":{"x":160.85000610351562,"y":393,"width":40,"height":43},"pins":[{"adj":[["41",0]]},{"adj":[["2","0"]]},{"adj":[["31",0]]}]},"31":{"nid":"31","type":"wire","pos":{"x":201,"y":450},"rect":{"x":193,"y":442,"width":16,"height":16},"pins":[{"adj":[["30","2"],["34",0]]}]},"33":{"nid":"33","type":"wire","pos":{"x":147,"y":472},"rect":{"x":139,"y":464,"width":16,"height":16},"pins":[{"adj":[["26","0"],["34",0],["44","2"]]}]},"34":{"nid":"34","type":"wire","pos":{"x":147,"y":450},"rect":{"x":139,"y":442,"width":16,"height":16},"pins":[{"adj":[["28","2"],["31","0"],["33","0"]]}]},"37":{"nid":"37","type":"nmos","pos":{"x":72.85000610351562,"y":321.00001525878906},"rect":{"x":72.85000610351562,"y":321.00001525878906,"width":40,"height":43},"pins":[{"adj":[["51",0]]},{"adj":[["6","0"]]},{"adj":[["42",0]]}]},"39":{"nid":"39","type":"nmos","pos":{"x":196.85000610351562,"y":311.99999237060547},"rect":{"x":196.85000610351562,"y":311.99999237060547,"width":40,"height":43},"pins":[{"adj":[["51",0]]},{"adj":[["7","0"]]},{"adj":[["40","0"]]}]},"40":{"nid":"40","type":"wire","pos":{"x":237,"y":365},"rect":{"x":229,"y":357,"width":16,"height":16},"pins":[{"adj":[["39","2"],["42",0]]}]},"41":{"nid":"41","type":"wire","pos":{"x":135,"y":395},"rect":{"x":127,"y":387,"width":16,"height":16},"pins":[{"adj":[["28","0"],["30","0"],["42",0]]}]},"42":{"nid":"42","type":"wire","pos":{"x":135,"y":365},"rect":{"x":127,"y":357,"width":16,"height":16},"pins":[{"adj":[["37","2"],["40","0"],["41",0]]}]},"44":{"nid":"44","type":"nmos","pos":{"x":316.8500061035156,"y":385.99999237060547},"rect":{"x":316.8500061035156,"y":385.99999237060547,"width":40,"height":43},"pins":[{"adj":[["49","2"]]},{"adj":[["12","0"]]},{"adj":[["33","0"]]}]},"49":{"nid":"49","type":"nmos","pos":{"x":302.8500061035156,"y":299.00000762939453},"rect":{"x":302.8500061035156,"y":299.00000762939453,"width":40,"height":43},"pins":[{"adj":[["52",0]]},{"adj":[["13","0"]]},{"adj":[["44","0"]]}]},"50":{"nid":"50","type":"wire","pos":{"x":415,"y":242},"rect":{"x":407,"y":234,"width":16,"height":16},"pins":[{"adj":[["52",0],["54",0]]}]},"51":{"nid":"51","type":"wire","pos":{"x":169,"y":313.99999237060547},"rect":{"x":161,"y":305.99999237060547,"width":16,"height":16},"pins":[{"adj":[["37","0"],["39","0"],["52",0]]}]},"52":{"nid":"52","type":"wire","pos":{"x":341,"y":242},"rect":{"x":333,"y":234,"width":16,"height":16},"pins":[{"adj":[["50","0"],["51","0"],["49","0"]]}]},"54":{"nid":"54","type":"wire","pos":{"x":577,"y":20},"rect":{"x":569,"y":12,"width":16,"height":16},"pins":[{"adj":[["24","0"],["25","0"],["50","0"]]}]},"55":{"nid":"55","type":"wire","pos":{"x":191,"y":33},"rect":{"x":183,"y":25,"width":16,"height":16},"pins":[{"adj":[["1","0"],["56",0]]}]},"56":{"nid":"56","type":"wire","pos":{"x":191,"y":49},"rect":{"x":183,"y":41,"width":16,"height":16},"pins":[{"adj":[["55","0"],["16","0"],["4","0"]]}]}},"inputPins":[{"name":"A"},{"name":"B"},{"name":"C"},{"name":"D"},{"name":"E"},{"name":"F"}],"outputPins":[{"name":"Y"}],"nNodes":58,"simType":"transistor","version":"7"}