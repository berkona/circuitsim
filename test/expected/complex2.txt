{ "inputs": [ "A", "B", "E", "F", "C", "D" ],
  "outputs": [ "Y" ],
  "rows": 
   [ [ "1", "1", "1", "1", "1", "1", "0" ],
     [ "1", "1", "1", "1", "1", "0", "0" ],
     [ "1", "1", "1", "1", "0", "1", "0" ],
     [ "1", "1", "1", "1", "0", "0", "0" ],
     [ "1", "1", "1", "0", "1", "1", "0" ],
     [ "1", "1", "1", "0", "1", "0", "0" ],
     [ "1", "1", "1", "0", "0", "1", "0" ],
     [ "1", "1", "1", "0", "0", "0", "0" ],
     [ "1", "1", "0", "1", "1", "1", "0" ],
     [ "1", "1", "0", "1", "1", "0", "0" ],
     [ "1", "1", "0", "1", "0", "1", "0" ],
     [ "1", "1", "0", "1", "0", "0", "0" ],
     [ "1", "1", "0", "0", "1", "1", "0" ],
     [ "1", "1", "0", "0", "1", "0", "1" ],
     [ "1", "1", "0", "0", "0", "1", "1" ],
     [ "1", "1", "0", "0", "0", "0", "1" ],
     [ "1", "0", "1", "1", "1", "1", "0" ],
     [ "1", "0", "1", "1", "1", "0", "0" ],
     [ "1", "0", "1", "1", "0", "1", "0" ],
     [ "1", "0", "1", "1", "0", "0", "0" ],
     [ "1", "0", "1", "0", "1", "1", "0" ],
     [ "1", "0", "1", "0", "1", "0", "0" ],
     [ "1", "0", "1", "0", "0", "1", "0" ],
     [ "1", "0", "1", "0", "0", "0", "0" ],
     [ "1", "0", "0", "1", "1", "1", "0" ],
     [ "1", "0", "0", "1", "1", "0", "0" ],
     [ "1", "0", "0", "1", "0", "1", "0" ],
     [ "1", "0", "0", "1", "0", "0", "0" ],
     [ "1", "0", "0", "0", "1", "1", "0" ],
     [ "1", "0", "0", "0", "1", "0", "1" ],
     [ "1", "0", "0", "0", "0", "1", "1" ],
     [ "1", "0", "0", "0", "0", "0", "1" ],
     [ "0", "1", "1", "1", "1", "1", "0" ],
     [ "0", "1", "1", "1", "1", "0", "0" ],
     [ "0", "1", "1", "1", "0", "1", "0" ],
     [ "0", "1", "1", "1", "0", "0", "0" ],
     [ "0", "1", "1", "0", "1", "1", "0" ],
     [ "0", "1", "1", "0", "1", "0", "0" ],
     [ "0", "1", "1", "0", "0", "1", "0" ],
     [ "0", "1", "1", "0", "0", "0", "0" ],
     [ "0", "1", "0", "1", "1", "1", "0" ],
     [ "0", "1", "0", "1", "1", "0", "0" ],
     [ "0", "1", "0", "1", "0", "1", "0" ],
     [ "0", "1", "0", "1", "0", "0", "0" ],
     [ "0", "1", "0", "0", "1", "1", "0" ],
     [ "0", "1", "0", "0", "1", "0", "1" ],
     [ "0", "1", "0", "0", "0", "1", "1" ],
     [ "0", "1", "0", "0", "0", "0", "1" ],
     [ "0", "0", "1", "1", "1", "1", "0" ],
     [ "0", "0", "1", "1", "1", "0", "1" ],
     [ "0", "0", "1", "1", "0", "1", "1" ],
     [ "0", "0", "1", "1", "0", "0", "1" ],
     [ "0", "0", "1", "0", "1", "1", "0" ],
     [ "0", "0", "1", "0", "1", "0", "1" ],
     [ "0", "0", "1", "0", "0", "1", "1" ],
     [ "0", "0", "1", "0", "0", "0", "1" ],
     [ "0", "0", "0", "1", "1", "1", "0" ],
     [ "0", "0", "0", "1", "1", "0", "1" ],
     [ "0", "0", "0", "1", "0", "1", "1" ],
     [ "0", "0", "0", "1", "0", "0", "1" ],
     [ "0", "0", "0", "0", "1", "1", "0" ],
     [ "0", "0", "0", "0", "1", "0", "1" ],
     [ "0", "0", "0", "0", "0", "1", "1" ],
     [ "0", "0", "0", "0", "0", "0", "1" ] ] }